exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -387,3 +409,25 @@ func (r *ClusterResourceSetReconciler) clusterToClusterResourceSet(o handler.Map
 	}
 	return result
 }
+
+// resourceToClusterResourceSet is mapper function that maps resources to ClusterResourceSet
+func (r *ClusterResourceSetReconciler) resourceToClusterResourceSet(o handler.MapObject) []ctrl.Request {
+	result := []ctrl.Request{}
+
+	crsList := &addonsv1.ClusterResourceSetList{}
+	if err := r.Client.List(context.Background(), crsList, client.InNamespace(o.Meta.GetNamespace())); err != nil {
+		return nil
+	}
+
+	for _, crs := range crsList.Items {
+		for _, resource := range crs.Spec.Resources {
+			if resource.Kind == o.Object.GetObjectKind().GroupVersionKind().Kind && crs.Name == o.Meta.GetName() {
```

Comment:
Technically we should be checking on both Group and Kind, as someone could create a foo.io/v123 Secret or ConfigMap.

Reply:
Do we need to worry about core vs empty group equivalency here when comparing the group, though? I think core types are probably the one exception where it might be ok to avoid checking group.

Reply:
I'm suggesting we worry about ensuring group="",Kind="Secret" and not matching on group="foo.io",Kind="Secret" - wouldn't that make sense?

Reply:
Do we need to worry about also testing for group == "core" in addition to group == "" ?

Reply:
No - despite being known as the core group, the only valid value for that group is ""

Reply:
`o.Object.GetObjectKind()` --> comes out empty `&TypeMeta{Kind:,APIVersion:,}`

I see the `Kind` when I print the object though (o.Object): `&Secret{ObjectMeta:{mySecret default /api/ ...`

Reply:
cc @vincepri @ncdc

Reply:
Can you print with `%#v` and report back?

Reply:
```
 &v1.ConfigMap{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"sedefjsonccc", GenerateName:"", Namespace:"default", SelfLink:"/api/v1/namespaces/default/configmaps/sedefjsonccc", UID:"417b0ebe-1304-4255-9b0e-45fda3d4543e", ResourceVersion:"1944", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:63731726962, loc:(*time.Location)(0x26fcbc0)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference(nil), Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, Data:map[string]string{"calico.json":"[{\n  \"kind\": \"ConfigMap\",\n  \"apiVersion\": \"v1\",\n  \"metadata\": {\n    \"name\": \"aug1\",\n    \"namespace\": \"default\"\n  }\n},\n{\n  \"kind\": \"ConfigMap\",\n  \"apiVersion\": \"v1\",\n  \"metadata\": {\n    \"name\": \"aug2\",\n    \"namespace\": \"default\"\n  }\n}\n]\n"}, BinaryData:map[string][]uint8(nil)}
```

Reply:
Thanks, given that TypeMeta is missing, you can either use reflection, or write `resourceToClusterResourceSet` such that it takes in the expected GroupKind and returns a map func that uses the input GroupKind in the comparison instead of checking data on `o`.

Reply:
If you store the `runtime.Scheme` on `r` (ClusterResourceSetReconciler), you should be able to use `apiutil.GVKForObject`

Reply:
It was originally there but I had Sedef remove it because we weren't using it 😄. Would it make more sense to do the func-that-returns-a-func approach so we don't have to do a lookup every time?

Reply:
This is the mapping function's parameter:

`func (r *ClusterResourceSetReconciler) resourceToClusterResourceSet(o handler.MapObject) []ctrl.Request {`

Cannot reach o's Kind.

Reply:
Yeah, `func createMapFunc(kind schema.GroupKind) func(o handler.MapObject) []ctrl.Request { return func(...) ....}`

Reply:
We removed the Scheme in the Binding controller, we still have it here, so used that one.